fuel/downloaders/base.py

```diff
@@ -117,6 +117,13 @@ def default_downloader(directory, urls, filenames, url_prefix=None,
                 os.remove(f)
     else:
         print('Downloading ' + ', '.join(filenames) + '\n')
+        try:
+            os.makedirs(directory)
+        except OSError as e:
```

Comment:
This one has a race condition in it (see e.g. http://stackoverflow.com/questions/18973418/os-mkdirpath-returns-oserror-when-directory-does-not-exist). Chances are low that it will happen here, but still it is best to not write fragile code